test/IDE/complete_name_lookup.swift

```diff
@@ -50,7 +50,7 @@ extension ObservableConvertibleType {
 // CATCHSEQUENCE_DOT: Begin completions
 // CATCHSEQUENCE_DOT-DAG: Keyword[self]/CurrNominal:          self[#CatchSequence<_>.Type#]; name=self
 // CATCHSEQUENCE_DOT-DAG: Keyword/CurrNominal:                Type[#CatchSequence<_>.Type#]; name=Type
-// CATCHSEQUENCE_DOT-DAG: Decl[Constructor]/CurrNominal:      init()[#CatchSequence<_>#]; name=init()
-// CATCHSEQUENCE_DOT-DAG: Decl[StaticMethod]/Super:           catchError()[#Observable<CatchSequence<_>.T>#]; name=catchError()
+// CATCHSEQUENCE_DOT-DAG: Decl[Constructor]/CurrNominal:      init()[#CatchSequence<Sequence>#]; name=init()
+// CATCHSEQUENCE_DOT-DAG: Decl[StaticMethod]/Super:           catchError()[#Observable<ObservableConvertibleType.T>#]; name=catchError()
```

Comment:
This seems to be a little regression. @slavapestov What do you think?

Reply:
Since the assertion is coming from substituting into a GenericFunctionType in particular, you could instead try a narrower workaround that only targets that case. Another approach is to tweak the GenericSignatureBuilder or the CanTypeMatcher to gracefully handle an UnresolvedType.

Reply:
Thanks Slava! I'll give those a try.